packages/sil-governance/src/routes/dashboard/components/ProposalsList.tsx

```diff
@@ -34,14 +38,45 @@ const getFilter = (filterType: ElectionFilter, currentUser: Address | null): Pro
   }
 };
 
-function compareById<E extends { readonly id: number }>(element1: E, element2: E): number {
-  return element1.id - element2.id;
-}
+type ProposalsComparator = (proposal1: ProposalProps, proposal2: ProposalProps) => number;
+
+const compareById: ProposalsComparator = (proposal1, proposal2): number => proposal1.id - proposal2.id;
+const compareByExpiryDate: ProposalsComparator = (proposal1, proposal2): number => {
+  return proposal1.expiryDate.getTime() - proposal2.expiryDate.getTime();
+};
+const compareByStartDate: ProposalsComparator = (proposal1, proposal2): number => {
+  return proposal1.startDate.getTime() - proposal2.startDate.getTime();
+};
+const compareByVote: ProposalsComparator = (proposal1, proposal2): number => {
+  if (proposal1.vote === undefined) return 1;
```

Comment:
This gives you `proposal1 > proposal2` and `proposal2 > proposal1` at the same time if `.vote` is undefined in both of them.